### PR TITLE
CEPH-83574190 Subvolume default and different permissions

### PR DESCRIPTION
**Pull request by julpark-rh**
Pre-requisites:
    1. Create cephfs subvolumegroup
    2. Create cephfs subvolume in subvolume group with default permission
    3. Create cephfs subvolumes in subvolumegroup with different permission
Test operation:
    1. Get path of all created subvolumes
    2. Remove all the subvolumes
    3. Remove the subvolumegroup
Regarding to permissions:

16895 = 777 (Different permission)
16895 = 755 (Default permission)

https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574190
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-S9KS2M/

Signed-off-by: julpark-rh <yonhyun@gmail.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        log.info(f"Running CephFS tests for Polarion ID -{tc}")
+        fs_util = FsUtils(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        client1 = clients[0]
+        fs_details = fs_util.get_fs_info(client1)
+        if not fs_details:
+            fs_util.create_fs(client1, "cephfs")
+        fs_util.auth_list([client1])
+        subvol_group = "subvolume_groupname_" + "".join(
+            random.choice(string.ascii_lowercase + string.digits)
```
Comment (AmarnatReddy) [inline tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py:33]: Can we generate random string once and use it
vol_name_rand =  random.choice(string.ascii_lowercase + string.digits)
            for _ in list(range(5))
subvol_group = f"subvolume_groupname_{vol_name_rand}"
subvol_default = f"subvol_default_{vol_name_rand}"
subvol_different = f"subvol_different_{vol_name_rand}"

File: tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        )
+        out4, err4 = client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath cephfs {subvol_different} --group_name {subvol_group}",
+        )
+        mode1 = out1["mode"]
+        mode2 = out2["mode"]
+        if str(mode1) != "16877" or str(mode2) != "16895" or out3 == 1 or out4 == 1:
+            return 1
+        fs_util.remove_subvolume(
```
Comment (AmarnatReddy) [inline tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py:69]: Can we move cleanup to finally block (not mandatory)

File: tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py (hunk 1 above)
Comment (julpark-rh) [reply tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_permissions.py]: @AmarnatReddy I have updated the changes.

**Review by AmarnatReddy — APPROVED** [review]
LTGM

**Pull request merged**